tests/python_package_test/test_dask.py

```diff
@@ -226,6 +374,54 @@ def test_regressor_local_predict(client, listen_port):
     assert_eq(p1, p2)
     assert_eq(s1, s2)
 
+    client.close()
+
+
+@pytest.mark.parametrize('output', ['array', 'dataframe'])
+@pytest.mark.parametrize('group', [None, group_sizes])
+def test_ranker(output, client, listen_port, group):
+
+    X, y, w, g, dX, dy, dw, dg = _create_ranking_data(output=output, group=group)
+
+    # -- use many trees + leaves to overfit, help ensure that dask data-parallel strategy matches that of
+    # -- serial learner. See https://github.com/microsoft/LightGBM/issues/3292#issuecomment-671288210.
+    dask_ranker = dlgbm.DaskLGBMRanker(time_out=5, local_listen_port=listen_port, tree_learner_type='data_parallel',
+                                       n_estimators=50, num_leaves=20, seed=42, min_child_samples=1)
+    dask_ranker = dask_ranker.fit(dX, dy, sample_weight=dw, group=dg, client=client)
+    rnkvec_dask = dask_ranker.predict(dX)
+    rnkvec_dask = rnkvec_dask.compute()
+
+    local_ranker = lightgbm.LGBMRanker(n_estimators=50, num_leaves=20, seed=42, min_child_samples=1)
+    local_ranker.fit(X, y, sample_weight=w, group=g)
+    rnkvec_local = local_ranker.predict(X)
+
+    # distributed ranker should be able to rank decently well and should
+    # have high rank correlation with scores from serial ranker.
+    dcor = spearmanr(rnkvec_dask, y).correlation
+    assert dcor > 0.6
+    assert spearmanr(rnkvec_dask, rnkvec_local).correlation > 0.9
+
+    client.close()
+
+
+@pytest.mark.parametrize('output', ['array', 'dataframe'])
+@pytest.mark.parametrize('group', [None, group_sizes])
+def test_ranker_local_predict(output, client, listen_port, group):
+
+    X, y, w, g, dX, dy, dw, dg = _create_ranking_data(output=output, group=group)
+
+    dask_ranker = dlgbm.DaskLGBMRanker(time_out=5, local_listen_port=listen_port, tree_learner='data',
+                                       n_estimators=10, num_leaves=10, seed=42, min_child_samples=1)
+    dask_ranker = dask_ranker.fit(dX, dy, group=dg, client=client)
+    rnkvec_dask = dask_ranker.predict(dX)
+    rnkvec_dask = rnkvec_dask.compute()
+    rnkvec_local = dask_ranker.to_local().predict(X)
+
+    # distributed and to-local scores should be the same.
+    assert_eq(rnkvec_dask, rnkvec_local)
```

Comment:
@jameslamb I like this test very much! Can we add the same for other estimators? Seems that Classifier tests do not asserts real Dask `predict()` equals to `.to_local().predict()`.

Reply:
the classifier tests do check that

* comparison between Dask `predict()` and non-Dask `predict()`: https://github.com/microsoft/LightGBM/blob/1c60cfa6a6dc8f24cfc2f37b0555cf1252366370/tests/python_package_test/test_dask.py#L94
* comparison between Dask `predict()` and `.to_local().predict()`: https://github.com/microsoft/LightGBM/blob/1c60cfa6a6dc8f24cfc2f37b0555cf1252366370/tests/python_package_test/test_dask.py#L138

I'd be open to a PR that makes it more explicit and cuts out the total number of tests by combining `test_classifier_local_predict` (https://github.com/microsoft/LightGBM/blob/1c60cfa6a6dc8f24cfc2f37b0555cf1252366370/tests/python_package_test/test_dask.py#L122) into `test_classifier` (https://github.com/microsoft/LightGBM/blob/1c60cfa6a6dc8f24cfc2f37b0555cf1252366370/tests/python_package_test/test_dask.py#L72) though!

Not an action we need on this PR, I'll write it up as a "good first issue".

Reply:
@jameslamb 
> comparison between Dask `predict()` and `to_local().predict()`:

Actually this checks `to_local().predict(dX)` and **local** `predict()`. It is not the same as `to_local().predict(dX)` and **Dask** `predict()`, given that asserts you pointing are placed in two different (in theory unrelated) tests, I think.

Reply:
it's confusing because it's two tests, so I think we should merge the tests in the future like I said above (not anything you need to do on here, @ffineis).

I'm saying that taken together, they show *transitively* that `dask_model.predict() == dask_model.to_local().predict()`, even though neither one does that direct comparison.

* (line 92) `dask_model.predict() == sklearn_model.predict()`
* (line 138) `dask_model.to_local().predict() == sklearn_model.predict()`

But that's only partially true, since `test_local_predict` only does binary classification, and only on array data. So combining them would give more coverage and would show whether that result holds up for multi-class classification or for other input data types.

Reply:
> it's confusing because it's two tests, so I think we should merge the tests in the future like I said above

Totally agree!

> I'm saying that taken together, they show *transitively* that `dask_model.predict() == dask_model.to_local().predict()`

Also agree. But two different tests can take different inputs (params, data), can be modified in future separately, etc. So it can be not fully fair comparison. Merging them into one test will solve all issues. 🙂